lib/Futures/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+add_library(arango_futures STATIC
+  Future.cpp)
+
+target_include_directories(arango_futures PRIVATE
+  ${PROJECT_SOURCE_DIR}/3rdParty/iresearch/external/function2
+  PUBLIC
+  ${PROJECT_SOURCE_DIR}/lib
```

Comment:
Do we really need this? In this PR you already changed all "Futures/...." includes into "...". If you need more, perhaps it needs to be linked? (see my comment below about the assertions)